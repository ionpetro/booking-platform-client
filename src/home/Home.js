import {gsap} from 'gsap';
import Button from '../components/Button/Button.vue';
import Recommendation from '../components/Recommendation/Recommendation.vue';
import Footer from '../components/Footer/Footer.vue';

export default {
  name: 'home',
  components: {
    'v-recommendation': Recommendation,
    'v-button': Button,
    Footer
  },
  data() {
    return {
      authenticated: false
    };
  },
  // this component is keep-alive, so we need to
  // check if user is authenticated every time it
  // is activated
  activated() {
    this.authenticated = this.isAuth();
  },
  methods: {
    isAuth() {
      return !!(localStorage.getItem('user') || sessionStorage.getItem('user'));
    },
    viewUnits() {
      if (this.authenticated) {
        this.$router.push('/units');
      } else {
        this.$router.push({name: 'login', query: {redirect: '/units'}});
      }
    },
    animateHomePage() {
      gsap.set('.animateIntro', {x: -50, opacity: 0});

      const tl = gsap.timeline({defaults: {duration: 1}});

      tl.from('.home__main__image', {x: 100}).to(
        '.animateIntro',
        {
          x: 0,
          opacity: 1,
          stagger: 0.4
        },
        '-=1'
      );
    }
  },
  mounted() {
    this.animateHomePage();
  }
};
