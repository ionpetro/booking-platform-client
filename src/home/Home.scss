$full-height: calc(100vh - 71px);

.home {
  overflow-x: hidden;

  &__recommendation {
    height: $full-height;
    display: flex;
    align-items: center;
  }

  &__main {
    height: $full-height;
    width: 100%;
    display: flex;
    overflow-x: hidden;
    justify-content: space-between;
    align-items: center;

    &__intro {
      width: 100%;
      display: flex;
      align-items: center;
      justify-content: center;

      &__title {
        font-weight: 500;
        font-size: 72px;
        line-height: 100px;

        @media only screen and (max-width: 1480px) {
          font-size: 60px;
          line-height: 60px;
        }

        @media #{$tablet} {
          font-size: 50px;
          letter-spacing: -0.005em;
        }

        @media #{$mobile} {
          font-size: 24px;
          line-height: 36px;
        }
      }
      &__button {
        margin-top: 46px;
        @media #{$mobile} {
          font-size: 12px;
        }
      }
    }

    &__image {
      float: right;
      max-height: 1300px;
      height: $full-height;

      @media only screen and (max-width: 1480px) {
        margin-right: -200px;
      }

      @media #{$tablet} {
        min-height: 700px;
        height: 80vh;
        margin-right: -400px;
      }

      @media #{$mobile} {
        height: 60vh;
        max-height: 400px;
        min-height: auto;
        margin-right: -200px;
      }
    }

    // TODO test with Marietta if 1680px works on her laptop

    @media only screen and (max-width: 1480px) {
      margin-left: 30px;
    }

    @media #{$tablet} {
      margin-left: 16px;
    }

    @media #{$mobile} {
      justify-content: flex-start;
    }
  }

  &__partnerships {
    text-align: center;

    &__h1 {
      @media #{$tablet} {
        font-size: 24px;
      }
      @media #{$mobile} {
        margin-bottom: 40px;
      }
    }

    &__images {
      display: flex;
      justify-content: center;

      &__image {
        margin: 0 40px;

        @media #{$tablet} {
          margin: auto;
          width: 130px;
        }
      }
      &__nasa {
        @media #{$tablet} {
          width: 70px;
        }
      }

      @media #{$mobile} {
        flex-direction: column;
      }
    }
  }
}
