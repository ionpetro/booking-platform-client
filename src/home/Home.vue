<template>
  <div class="home">
    <v-recommendation v-if="authenticated" class="home__recommendation"></v-recommendation>
    <section class="home__main">
      <div class="home__main__intro">
        <div>
          <div class="home__main__intro__title animateIntro">
            We are giving you <br />
            the space you need
          </div>
          <v-button
            class="home__main__intro__button animateIntro"
            @click.native="viewUnits"
            :variant="'secondary'"
            >Explore units</v-button
          >
        </div>
      </div>
      <div>
        <img class="home__main__image" src="@/assets/images/Units.png" alt="Units" />
      </div>
    </section>
    <section class="home__partnerships">
      <h1 class="home__partnerships__h1">Partnerships</h1>
      <div class="home__partnerships__images">
        <img
          class="home__partnerships__images__image home__partnerships__images__nasa"
          src="@/assets/images/Nasa.png"
          alt="Nasa"
        />
        <img
          class="home__partnerships__images__image"
          src="@/assets/images/SpaceX.png"
          alt="SpaceX"
        />
        <img class="home__partnerships__images__image" src="@/assets/images/Eesa.png" alt="Eesa" />
        <img
          class="home__partnerships__images__image"
          src="@/assets/images/ArianeGroup.png"
          alt="ArianeGroup"
        />
      </div>
    </section>
    <Footer></Footer>
  </div>
</template>

<script src="./Home.js"></script>

<style lang="scss" src="./Home.scss" scoped></style>
