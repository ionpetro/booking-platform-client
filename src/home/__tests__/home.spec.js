import {shallowMount} from '@vue/test-utils';
import flushPromises from 'flush-promises';
import Home from '../Home.vue';

describe('Home', () => {
  let wrapper;
  const $router = {push: jest.fn()};

  beforeEach(() => {
    wrapper = shallowMount(Home, {
      mocks: {$router}
    });
  });

  describe('when loaded', () => {
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('redirects to units on action button click', async () => {
      // when user is logged in
      wrapper.vm.$data.authenticated = true;

      await wrapper.find('.home__main__intro__button').trigger('click');

      await flushPromises();

      expect($router.push).toHaveBeenCalledWith('/units');
    });

    it('redirects to units with params on action button click', async () => {
      // when user is not logged in
      wrapper.vm.$data.authenticated = false;

      await wrapper.find('.home__main__intro__button').trigger('click');

      await flushPromises();

      expect($router.push).toHaveBeenCalledWith({name: 'login', query: {redirect: '/units'}});
    });
  });
});
