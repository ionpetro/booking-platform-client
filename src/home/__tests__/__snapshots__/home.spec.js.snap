// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home when loaded renders correctly 1`] = `
<div
  class="home"
>
  <!---->
   
  <section
    class="home__main"
  >
    <div
      class="home__main__intro"
    >
      <div>
        <div
          class="home__main__intro__title animateIntro"
        >
          
          We are giving you 
          <br />
          
          the space you need
        
        </div>
         
        <v-button-stub
          class="home__main__intro__button animateIntro"
          variant="secondary"
        >
          Explore units
        </v-button-stub>
      </div>
    </div>
     
    <div>
      <img
        alt="Units"
        class="home__main__image"
        src="@/assets/images/Units.png"
      />
    </div>
  </section>
   
  <section
    class="home__partnerships"
  >
    <h1
      class="home__partnerships__h1"
    >
      Partnerships
    </h1>
     
    <div
      class="home__partnerships__images"
    >
      <img
        alt="Nasa"
        class="home__partnerships__images__image home__partnerships__images__nasa"
        src="@/assets/images/Nasa.png"
      />
       
      <img
        alt="SpaceX"
        class="home__partnerships__images__image"
        src="@/assets/images/SpaceX.png"
      />
       
      <img
        alt="Eesa"
        class="home__partnerships__images__image"
        src="@/assets/images/Eesa.png"
      />
       
      <img
        alt="ArianeGroup"
        class="home__partnerships__images__image"
        src="@/assets/images/ArianeGroup.png"
      />
    </div>
  </section>
   
  <footer-stub />
</div>
`;
